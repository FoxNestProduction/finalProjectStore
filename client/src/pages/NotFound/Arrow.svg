<svg width="167" height="167" viewBox="0 0 167 167" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M32.5541 95.566C48.9249 108.066 68.3193 116.347 89.539 112.624C108.692 109.264 121.366 95.604 109.568 76.9172C103.983 68.0713 91.4478 61.3162 81.0902 65.471C69.551 70.0998 77.6996 84.2167 84.6958 88.5913C101.036 98.8088 124.946 96.117 137.538 79.1664C152.45 59.0929 129.676 47.1295 129.651 47.2987" stroke="#6C5FBC" stroke-width="2.9437" stroke-miterlimit="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M35.0116 118.38C35.7449 116.224 35.5063 114.001 35.382 111.771C34.9747 104.447 32.1951 96.6906 26.7401 91.5488C34.0953 94.9903 44.8045 96.9793 51.9038 92.2451" stroke="#6C5FBC" stroke-width="2.9437" stroke-miterlimit="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
