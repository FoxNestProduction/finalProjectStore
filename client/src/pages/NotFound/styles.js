export const stylesWrapper = {
  position: 'relative',
  display: 'flex',
  flexDirection: 'column',
  height: '80vh',
  alignItems: 'center',
  justifyContent: 'center',
  textAlign: 'center',
  transition: '.3s',
};

export const stylesStatusWrapper = {
  display: 'flex',
  justifyContent: 'space-around',
  width: '100%',
};
export const stylesStatus = {
  fontSize: {
    mobile: '100px',
    tablet: '140px',
    desktop: '180px',
  },
  color: 'primary.main',
  fontWeight: 'fontWeightSemiBold',
  fontFamily: 'fontPoppins',
  lineHeight: {
    mobile: '64px',
    tablet: '130px',
    desktop: '150px',
  },
  marginBottom: {
    mobile: '40px',
    tablet: '60px',
  },
};

export const stylesDescription = {
  fontSize: {
    mobile: '16px',
    tablet: '20px',
    desktop: '24px',
  },
  color: 'text.primary',
  fontWeight: 'fontWeightSemiBold',
  fontFamily: 'fontPoppins',
  lineHeight: {
    mobile: '24px',
    tablet: '28px',
    desktop: '32px',
  },
  marginBottom: {
    mobile: '47px',
    tablet: '60px',
  },
};

export const stylesLink = {
  backgroundColor: 'primary.main',
  '&:hover': {
    backgroundColor: 'primary.hover',
  },
  '&:active': {
    backgroundColor: 'background.default',
    border: '1px solid #6C5FBC',
    color: 'primary.main',
  },
  height: '60px',
  padding: {
    mobile: '17px 55px',
    tablet: '17px 80px',
    desktop: '17px 100px',
  },
  fontSize: {
    mobile: '16px',
    tablet: '20px',
    desktop: '24px',
  },
  color: 'text.primaryLight',
  fontWeight: 500,
  fontFamily: 'fontPoppins',
  lineHeight: {
    mobile: '16px',
    tablet: '20px',
    desktop: '24px',
  },
};

export const stylesRightArrow = {
  position: 'relative',
  top: '-10%',
  width: '122px',
  display: {
    mobile: 'none',
    tablet: 'block',
  },
  transform: 'scaleX(-1)',
  rotate: '60deg',
};

export const stylesLeftArrow = {
  ...stylesRightArrow,
  transform: 'scale(-1, -1)',
  rotate: '120deg',
};
