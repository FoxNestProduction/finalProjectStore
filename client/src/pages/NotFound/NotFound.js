import React, { memo } from 'react';
import { Button, Typography } from '@mui/material';
import { Link as RouterLink } from 'react-router-dom';
import Container from '@mui/material/Container';
import Box from '@mui/material/Box';
import { stylesWrapper, stylesStatus, stylesDescription, stylesLink, stylesLeftArrow, stylesRightArrow, stylesStatusWrapper } from './styles';
import OneLoopArrowSvg from '../../assets/svgComponents/OneLoopArrowSvg';

const NotFound = () => (
  <Container component="div" sx={stylesWrapper}>
    <Box sx={stylesStatusWrapper}>
      <Box sx={stylesLeftArrow}><OneLoopArrowSvg /></Box>
      <Typography sx={stylesStatus} variant="h4">404</Typography>
      <Box sx={stylesRightArrow}><OneLoopArrowSvg /></Box>
    </Box>
    <Typography sx={stylesDescription} variant="body1">
      Oops, something went wrong!
      <br />
      The page you&apos;re looking for doesn&apos;t exist.
    </Typography>
    <Button
      sx={stylesLink}
      component={RouterLink}
      to="/"
    >
      Go To Home Page
    </Button>
  </Container>
);

export default memo(NotFound);
