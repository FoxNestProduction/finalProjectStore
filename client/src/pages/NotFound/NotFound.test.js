import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NotFound from './NotFound';

describe('NotFound Pages', () => {
  test('should NotFound Pages', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});