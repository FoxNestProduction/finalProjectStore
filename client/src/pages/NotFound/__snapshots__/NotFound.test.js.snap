// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Pages should NotFound Pages 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-14ftzfr-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-kr4k96"
    >
      <div
        class="MuiBox-root css-1nq4wyu"
      >
        <svg
          fill="none"
          height="118"
          viewBox="0 0 117 118"
          width="117"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M57.9545 94.9949C47.2252 85.339 38.8069 73.1896 38.0046 58.5059C37.2806 45.2521 44.3231 35.2379 58.4214 41.1439C65.0953 43.9396 71.4587 51.6393 70.3128 59.0609C69.0362 67.3292 58.5027 63.3819 54.5536 59.1501C45.3299 49.2665 43.4373 32.8851 52.6569 22.5675C63.5751 10.3491 74.9329 24.3724 74.8255 24.4077"
            stroke="#6C5FBC"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="1.5"
            stroke-width="2.09329"
          />
          <path
            d="M42.5719 95.8242C43.8782 95.0961 45.3767 95.0149 46.8626 94.8559C51.7416 94.3332 57.2693 95.3597 61.4869 98.4709C58.0966 93.8961 55.1476 86.9063 57.1738 81.6141"
            stroke="#6C5FBC"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="1.5"
            stroke-width="2.09329"
          />
        </svg>
      </div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1b4ci2b-MuiTypography-root"
      >
        404
      </h4>
      <div
        class="MuiBox-root css-1f3n5sq"
      >
        <svg
          fill="none"
          height="118"
          viewBox="0 0 117 118"
          width="117"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M57.9545 94.9949C47.2252 85.339 38.8069 73.1896 38.0046 58.5059C37.2806 45.2521 44.3231 35.2379 58.4214 41.1439C65.0953 43.9396 71.4587 51.6393 70.3128 59.0609C69.0362 67.3292 58.5027 63.3819 54.5536 59.1501C45.3299 49.2665 43.4373 32.8851 52.6569 22.5675C63.5751 10.3491 74.9329 24.3724 74.8255 24.4077"
            stroke="#6C5FBC"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="1.5"
            stroke-width="2.09329"
          />
          <path
            d="M42.5719 95.8242C43.8782 95.0961 45.3767 95.0149 46.8626 94.8559C51.7416 94.3332 57.2693 95.3597 61.4869 98.4709C58.0966 93.8961 55.1476 86.9063 57.1738 81.6141"
            stroke="#6C5FBC"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="1.5"
            stroke-width="2.09329"
          />
        </svg>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-nvgiei-MuiTypography-root"
    >
      Oops, something went wrong!
      <br />
      The page you're looking for doesn't exist.
    </p>
    <a
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1iwedua-MuiButtonBase-root-MuiButton-root"
      href="/"
      tabindex="0"
    >
      Go To Home Page
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</DocumentFragment>
`;
